cli/command/stack/kubernetes/list.go

```diff
@@ -1,44 +1,33 @@
 package kubernetes
 
 import (
-	"sort"
-
+	"github.com/docker/cli/cli/command"
 	"github.com/docker/cli/cli/command/formatter"
 	"github.com/docker/cli/cli/command/stack/options"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"vbom.ml/util/sortorder"
 )
 
-// RunList is the kubernetes implementation of docker stack ls
-func RunList(dockerCli *KubeCli, opts options.List) error {
-	stacks, err := getStacks(dockerCli, opts.AllNamespaces)
+// GetStacks lists the kubernetes stacks
+func GetStacks(dockerCli command.Cli, opts options.List, kopts Options) ([]*formatter.Stack, error) {
+	kubeCli, err := WrapCli(dockerCli, kopts)
 	if err != nil {
-		return err
-	}
-	format := opts.Format
-	if format == "" || format == formatter.TableFormatKey {
-		format = formatter.KubernetesStackTableFormat
+		return nil, err
 	}
-	stackCtx := formatter.Context{
-		Output: dockerCli.Out(),
-		Format: formatter.Format(format),
+	if opts.AllNamespaces || len(opts.Namespaces) == 0 {
+		if dockerCli.ClientInfo().HasAll() {
+			opts.AllNamespaces = true
+		}
+		return getStacks(kubeCli, opts)
 	}
-	sort.Sort(byName(stacks))
-	return formatter.StackWrite(stackCtx, stacks)
+	return getStacksWithNamespaces(kubeCli, opts)
 }
 
-type byName []*formatter.Stack
-
-func (n byName) Len() int           { return len(n) }
-func (n byName) Swap(i, j int)      { n[i], n[j] = n[j], n[i] }
-func (n byName) Less(i, j int) bool { return sortorder.NaturalLess(n[i].Name, n[j].Name) }
-
-func getStacks(kubeCli *KubeCli, allNamespaces bool) ([]*formatter.Stack, error) {
+func getStacks(kubeCli *KubeCli, opts options.List) ([]*formatter.Stack, error) {
 	composeClient, err := kubeCli.composeClient()
 	if err != nil {
 		return nil, err
 	}
-	stackSvc, err := composeClient.Stacks(allNamespaces)
+	stackSvc, err := composeClient.Stacks(opts.AllNamespaces)
```

Comment:
Not for this PR, but we should consider passing an `opts` here, instead of a boolean (`opts.AllNamespaces`) to prevent future bloat in the signature